core/src/main/java/io/grpc/internal/RetriableStream.java

```diff
@@ -646,16 +646,32 @@ public final Attributes getAttributes() {
 
   @Override
   public void appendTimeoutInsight(InsightBuilder insight) {
+    State currentState;
     synchronized (lock) {
-      insight.appendKeyValue("finished", closedSubstreamsInsight);
+      insight.appendKeyValue("closed", closedSubstreamsInsight);
+      currentState = state;
     }
-    InsightBuilder drainedSubstreamsInsight = new InsightBuilder();
-    for (Substream sub : state.drainedSubstreams) {
+    if (currentState.winningSubstream != null) {
+      // TODO(zhangkun83): in this case while other drained substreams have been cancelled in favor
```

Comment:
This is not a todo.  There is nothing in this comment explaining what is to be done.